File: web/src/main/kotlin/com/isystk/sample/web/base/controller/api/resource/ErrorResourceImpl.java
package com.isystk.sample.web.base.controller.api.resource;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ErrorResourceImpl extends ResourceImpl {

  // リクエストID
  String requestId;

  // 入力エラー
  List<FieldErrorResource> fieldErrors;

  public ErrorResourceImpl() {
  }

  public String getRequestId() {
    return this.requestId;
  }

  public List<FieldErrorResource> getFieldErrors() {
    return this.fieldErrors;
  }

  public void setRequestId(String requestId) {
    this.requestId = requestId;
  }

  public void setFieldErrors(List<FieldErrorResource> fieldErrors) {
    this.fieldErrors = fieldErrors;
  }

  public boolean equals(final Object o) {
    if (o == this) {
      return true;
    }
    if (!(o instanceof ErrorResourceImpl)) {
      return false;
    }
    final ErrorResourceImpl other = (ErrorResourceImpl) o;
    if (!other.canEqual((Object) this)) {
      return false;
    }
    if (!super.equals(o)) {
      return false;
    }
    final Object this$requestId = this.getRequestId();
    final Object other$requestId = other.getRequestId();
    if (this$requestId == null ? other$requestId != null
        : !this$requestId.equals(other$requestId)) {
      return false;
    }
    final Object this$fieldErrors = this.getFieldErrors();
    final Object other$fieldErrors = other.getFieldErrors();
    if (this$fieldErrors == null ? other$fieldErrors != null
        : !this$fieldErrors.equals(other$fieldErrors)) {
      return false;
    }
    return true;
  }

  protected boolean canEqual(final Object other) {
    return other instanceof ErrorResourceImpl;
  }

  public int hashCode() {
    final int PRIME = 59;
    int result = super.hashCode();
    final Object $requestId = this.getRequestId();
    result = result * PRIME + ($requestId == null ? 43 : $requestId.hashCode());
    final Object $fieldErrors = this.getFieldErrors();
    result = result * PRIME + ($fieldErrors == null ? 43 : $fieldErrors.hashCode());
    return result;
  }
}

File: web/src/main/kotlin/com/isystk/sample/web/base/controller/api/resource/ErrorResourceImpl.kt
package com.isystk.sample.web.base.controller.api.resource

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
class ErrorResourceImpl : ResourceImpl() {
    // リクエストID
    var requestId: String? = null

    // 入力エラー
    var fieldErrors: List<FieldErrorResource>? = null
    override fun equals(o: Any?): Boolean {
        if (o === this) {
            return true
        }
        if (o !is ErrorResourceImpl) {
            return false
        }
        val other = o
        if (!other.canEqual(this as Any)) {
            return false
        }
        if (!super.equals(o)) {
            return false
        }
        val `this$requestId`: Any? = requestId
        val `other$requestId`: Any? = other.requestId
        if (if (`this$requestId` == null) `other$requestId` != null else `this$requestId` != `other$requestId`) {
            return false
        }
        val `this$fieldErrors`: Any? = fieldErrors
        val `other$fieldErrors`: Any? = other.fieldErrors
        return if (if (`this$fieldErrors` == null) `other$fieldErrors` != null else `this$fieldErrors` != `other$fieldErrors`) {
            false
        } else true
    }

    override fun canEqual(other: Any?): Boolean {
        return other is ErrorResourceImpl
    }

    override fun hashCode(): Int {
        val PRIME = 59
        var result = super.hashCode()
        val `$requestId`: Any? = requestId
        result = result * PRIME + (`$requestId`?.hashCode() ?: 43)
        val `$fieldErrors`: Any? = fieldErrors
        result = result * PRIME + (`$fieldErrors`?.hashCode() ?: 43)
        return result
    }
}